src/ctapipe/io/dl2_tables_preprocessing.py

```diff
@@ -0,0 +1,474 @@
+            ),
+            ("valid classifier", "RandomForestClassifier_is_valid"),
+            ("valid geom reco", "HillasReconstructor_is_valid"),
+            ("valid energy reco", "RandomForestRegressor_is_valid"),
+        ],
+        help=QualityQuery.quality_criteria.help,
+    ).tag(config=True)
+
+
+class DL2EventPreprocessor(Component):
+    """Defines pre-selection cuts and the necessary renaming of columns."""
+
+    classes = [DL2EventQualityQuery]
+
+    energy_reconstructor = Unicode(
+        default_value="RandomForestRegressor",
+        help="Prefix of the reco `_energy` column",
+    ).tag(config=True)
+
+    geometry_reconstructor = Unicode(
+        default_value="HillasReconstructor",
+        help="Prefix of the `_alt` and `_az` reco geometry columns",
+    ).tag(config=True)
+
+    gammaness_classifier = Unicode(
+        default_value="RandomForestClassifier",
+        help="Prefix of the classifier `_prediction` column",
+    ).tag(config=True)
+
+    apply_derived_columns = Bool(
+        default_value=True, help="Whether to compute derived columns"
+    ).tag(config=True)
+
+    apply_check_pointing = Bool(
+        default_value=True, help="Accept only pointing in altaz and not divergent."
+    ).tag(config=True)
+
+    columns_to_rename = Dict(
+        key_trait=Unicode(),
+        value_trait=Unicode(),
+        help=(
+            "Dictionary of columns to rename. "
+            "Leave unset to apply default renaming. "
+            "Set to an empty dictionary to disable renaming entirely. "
+            "Set to a partial dictionary to override only some names."
+        ),
+    ).tag(config=True)
+
+    output_table_schema = List(
+        default_value=[
+            Column(name="obs_id", dtype=np.uint64, description="Observation Block ID"),
+            Column(name="event_id", dtype=np.uint64, description="Array event ID"),
+            Column(name="true_energy", unit=u.TeV, description="Simulated energy"),
+            Column(name="true_az", unit=u.deg, description="Simulated azimuth"),
+            Column(name="true_alt", unit=u.deg, description="Simulated altitude"),
+            Column(name="reco_energy", unit=u.TeV, description="Reconstructed energy"),
+            Column(name="reco_az", unit=u.deg, description="Reconstructed azimuth"),
+            Column(name="reco_alt", unit=u.deg, description="Reconstructed altitude"),
+            Column(name="pointing_az", unit=u.deg, description="Pointing azimuth"),
+            Column(name="pointing_alt", unit=u.deg, description="Pointing altitude"),
+        ],
+        help="Schema definition for output event QTable",
+    ).tag(config=True)
+
+    def __init__(self, config=None, parent=None, **kwargs):
+        super().__init__(config=config, parent=parent, **kwargs)
+        self.quality_query = DL2EventQualityQuery(parent=self)
+
+    @default("columns_to_rename")
+    def _default_columns_to_rename(self):
+        return {
```

Comment:
A comment I had in the original PR for this as well: this renaming is really just a stop-gap hack until we have a proper Component to do the DL2→DL3 algorithm choice selection.  It was always in the design that we would be able to select per-event, not just per method, which columns to use for the DL3 output, e.g. which reconstructor to make the "final" one.  As currently implemented, this only lets you chose a single reconstruction.   Maybe we should at least add a comment that this is a temporary solution?